File: 1796_Second_Largest_Digit_in_a_String/main.go
import "sort"

func secondHighest(s string) int {
	digits := []int{-1}
	for _, v := range s {
		num, err := strconv.Atoi(string(v))
		if err == nil {
			digits = append(digits, num)
		}
	}

	sort.Ints(digits)
	max := digits[len(digits)-1]
	for i := len(digits) - 1; i >= 0; i-- {
		if digits[i] < max {
			return digits[i]
		}
	}
	return -1
}

File: 1796_Second_Largest_Digit_in_a_String/main.py
class Solution:
    def secondHighest(self, s: str) -> int:
        digits = set()
        for i in s:
            if i.isdigit():
                digits.add(int(i))

        digits = sorted(digits)
        if len(digits) > 1:
            return digits[-2]
        return -1
